File: ruby/hamming/hamming.rb
class BookKeeping
    VERSION = 3
  end
  
class Hamming
    def self.compute(original, mutation)
 
        raise ArgumentError unless original.size == mutation.size

        original.split("")
                .zip(mutation.split(""))
                .select { |x| x[0] != x[1] }
                .length
 
    end
end

File: javascript/hamming/hamming.js
                
class Hamming {
  
    compute(original, mutation) {
        if (original.length != mutation.length) throw new Error('DNA strands must be of equal length.');
    
        return original.split('').filter((x,i) => {
            let m = mutation.split('');
            return x != m[i];
        }).length;
    }
  
}

module.exports = Hamming;